src/Compilers/Core/MSBuildTask/Microsoft.CSharp.Core.targets

```diff
@@ -3,8 +3,8 @@
 <Project ToolsVersion="14.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <Import Project="Microsoft.Managed.Core.targets"/>
 
-  <PropertyGroup Condition="('$(TargetFrameworkIdentifier)' != '.NETCoreApp' OR '$(TargetFrameworkVersion)' != 'v3.0') AND
-                            ('$(TargetFrameworkIdentifier)' != '.NETStandard' OR '$(TargetFrameworkVersion)' != 'v2.1')">
+  <PropertyGroup Condition="('$(TargetFrameworkIdentifier)' != '.NETCoreApp' OR '$(_TargetFrameworkVersionWithoutV)' < '3.0') AND
```

Comment:
Is this doing a string comparison?

Reply:
You mean the `<` operator? I'm reasonably sure this has been updated in MSBuild to do a real version comparison, but I'd like @nguerrera to confirm that.

Reply:
+ @rainersigwald 

The answer here is a little convoluted. MSBuild can coerce to version here, but might coerce to float first. We actually have some latent bugs if there's ever a .10+ minor release of a TFM. :(

I would like there to be an intrinsic function that can operate directly on $(TargetFrameworkVersion) (v and all) and compare as we expect in all cases. This is tracked by https://github.com/microsoft/msbuild/issues/3212

Reply:
Nick is correct; `float` is a higher precedence than `Version`. And you can't just add `.0` to force to a version, because according to `System.Version`, [`3.0 < 3.0.0` is `true`](https://sharplab.io/#v2:C4LgTgrgdgNAJiA1AHwAICYCMBYAUKgZgAIMiBhIgbzyNpPXSprpYDcBDMI4ACzAFN+RALxEo/AO4lMAOgBq/MAGcAlgHsoACgBEBGQAZtASgDczFrQ5deA/gAU1KqMABaitSLGSiC5eq26BgbGZrgWdKiYAJyaNoJEADzcfIIOTq7upuYWkTFxQgB8ybZpzm5gallh4dJ5KULCovmlGRVVLAC+eB1AA). MSBuild's version comparison just wraps that so has the same funky behavior.

Fortunately, here you're breaking on an integer boundary so this implementation is fine (it's irrelevant for these purposes that `2.10 < 2.9` because they're both `<3.0`).

The right fix is that to-be-written property function but you wouldn't be able to use it for a long time due to the compiler package working on older MSBuild engines.

Reply:
I think I’ll take a stab at writing that function. Get the clock started at least. Moving forward I’d prefer to say “when you can depend on msbuild X, do this”